allauth_2fa/adapter.py

```diff
@@ -46,12 +47,12 @@ def get_2fa_authenticate_url(self, request: HttpRequest) -> str:
 
         # Add "next" parameter to the URL if possible.
         # If the view function smells like a class-based view, we can interrogate it.
-        if getattr(request.resolver_match.func, "view_class", None):
+        if get_next_query_string(request):
             view = request.resolver_match.func.view_class()
             view.request = request
             success_url = view.get_success_url()
             query_params = request.GET.copy()
-            if success_url:
+            if success_url and hasattr(view, "redirect_field_name"):
                 query_params[view.redirect_field_name] = success_url
             if query_params:
                 redirect_url += f"?{urlencode(query_params)}"
```

Comment:
Erm, shouldn't this just be

```suggestion
        query_string = get_next_query_string(request)
        if query_string:
            redirect_url += query_string
```

now that the logic is refactored..?

Reply:
Feel free to take this over! I don't use this library anymore and was just trying to pass this change forwards since it seems many of your users need it.